### DISCUSSION TITLE
Use of undefined constant STDIN - assumed 'STDIN' (this will throw an Error in a future version of PHP

### DISCUSSION THREAD

**Discussion by wmullaney** [composer/composer#10570] [Q&A / Support]
I am on a fresh aws cpanel build and can't get composer to create a project. I have to pull an older version of composer from another server to get things working.

**Comment by Seldaek** [comment]
Not sure what to tell you, I don't see what would be using STDIN before checking for its existence in the latest Composer version.. So you'd need to provide more details and possibly a stack trace if you want help.

**Reply by atimorales** [reply]
I have the same problem when I trying to install the mercado pago SDK with composer.
In puTTY (root user):
composer require "mercadopago/dx-php:1.12.5"
-->
Continue as root/super user [yes]? Use of undefined constant STDIN - assumed 'STDIN'
(like 100 times and after that "Maximum execution time of 30 seconds exceeded"

**Reply by fredden** [reply]
@atimorales what version of Composer are you using? Do you have any Composer plug-ins installed globally?

**Reply by atimorales** [reply]
I can resolve the problem!!
Look at this link: 
https://www.transip.eu/knowledgebase/entry/3300-installing-composer-in-linux/

First, I needed to use php-cli file (not only "php"), and install this libraries:
yum -y install php-cli php-zip wget unzip
After that, I can installa and use composer (the last version today) and install the mercado pago's SDK.
